### ISSUE TITLE
bug: stencil test with `e2e` trigger documentation build

### ISSUE THREAD

**Issue by Sukaato** [ionic-team/stencil#5950]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

4.20.0

### Current Behavior

When running `stencil test --e2e --spec`, stencil have to make a build for the e2e part, and this trigger generate the custom documentation, and generate the documentation should not be part of the testing phase.

### Expected Behavior

do not generate the custom documentation

### System Info

```shell
System: node 20.16.0
    Platform: linux (5.15.153.1-microsoft-standard-WSL2)
   CPU Model: AMD Ryzen 7 5700X 8-Core Processor (16 cpus)
    Compiler: /home/sukaato/projects/poppy-ui/node_modules/@stencil/core/compiler/stencil.js
       Build: 1722623140
     Stencil: 4.20.0 🚐
  TypeScript: 5.5.3
      Rollup: 2.56.3
      Parse5: 7.1.2
      jQuery: 4.0.0-pre
      Terser: 5.31.1
```

### Steps to Reproduce

1. Create component and generate the docs and commit
2. Update the component jsdocs so i can be highlighted on the next commit
3. run `npm test`  hould be by default stencil test --e2e --spec)*
4. See the readme has been updated

### Code Reproduction URL

https://github.com/Sukaato/stencil-test-generate-docs

### Additional Information

Should add a condition here: https://github.com/ionic-team/stencil/blob/main/src/compiler/output-targets/output-custom.ts#L7

Since `outputCustom` can generate more that a readme documentation, i think the flag `config.buildDocs` should be not used, maybe the `config._isTesting` can be used

**Comment by christian-bromann** [comment]
Thanks for reporting @Sukaato. I ingested the bug into our backlog but I want to set the expectations straight: we can't tell you when we will be able to tackle the issue. If you are blocked on it, we recommend to take a stab and provide a bug fix, the Stencil team is happy to provide guidance on that. Thanks!

**Comment by christian-bromann** [comment]
A fix for this has been released in Stencil `v4.22.0` 🎉
